File: src/main/java/com/ceaser/thrift/PersonServiceImpl.java
package com.ceaser.thrift;

import org.apache.thrift.TException;
import thrift.generated.DataException;
import thrift.generated.Person;
import thrift.generated.PersonService;

/**
 * Created by Administrator on 2017/6/10.
 */
public class PersonServiceImpl implements PersonService.Iface {
    @Override
    public Person getPersonByUserName(String username) throws DataException, TException {
        System.out.println("getPersonByUserName - get client param  : "+username);
        Person person = new Person();
        person.setAge(20);
        person.setMarried(true);
        person.setUsername("张三");
        return  person;
    }

    @Override
    public void savePerson(Person person) throws DataException, TException {
        System.out.println("savePerson - get client param  : ");
        System.out.println(person.getUsername());
        System.out.println(person.getAge());
        System.out.println(person.isMarried());
    }
}

File: pythrift/PersonServiceImpl.py
# coding=utf-8
__author='Ceaserwang'

from py.thrift.generated import ttypes

class PersonServiceImpl:
    def getPersonByUserName(self,username):
        print('get client param :'+username)
        person = ttypes.Person()
        person.married = True
        person.username=username
        person.age = 24
        return person

    def savePerson(self,person):
        print('get client param : ')
        print(person.username)
        print(person.age)
        print(person.married)
